tools/walletextension/wallet_extension.go

```diff
@@ -346,3 +352,45 @@ func adjustStateRoot(rpcResp interface{}, respMap map[string]interface{}) {
 		}
 	}
 }
+
+// handleSpecialRPCMethods checks method name in the request and if it is one of the methods that we use differently than
+// eth standard. At the moment it is only getStorageAt method. It returns response if we should use that response or nil otherwise
+func (w *WalletExtension) handleSpecialRPCMethods(request *accountmanager.RPCRequest, hexUserID string) map[string]interface{} {
+	response := map[string]interface{}{}
+	// all responses must contain the request id. Both successful and unsuccessful.
+	response[common.JSONKeyRPCVersion] = jsonrpc.Version
+	response[common.JSONKeyID] = request.ID
+
+	// check if requested method is GetStorageAt - as we use it for different purpose here with specific parameters
+	if request.Method == rpc.GetStorageAt && validateGetStorageAtForUserIDParams(request.Params) {
+		// check if userID in the parameters is also in our database
+		userID, err := common.GetUserIDbyte(hexUserID)
+		if err != nil {
+			return nil
```

Comment:
I'd also log the error here. To catch unexpected conditions

Reply:
fixed